File: app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.widget.Button;
import android.widget.VideoView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    VideoView videoView;
    Button startButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        videoView = findViewById(R.id.videoView);
        startButton = findViewById(R.id.button);

        // ▶ Video Background
        Uri videoUri = Uri.parse("android.resource://" + getPackageName() + "/" + R.raw.mm);
        videoView.setVideoURI(videoUri);

        videoView.setOnPreparedListener(mp -> {
            mp.setLooping(true);
            mp.setVolume(0f, 0f);
            videoView.start();
        });

        // ▶ START button click → Welcome page
        startButton.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, Welcome.class);
            startActivity(intent);
            overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
        });
    }

    @Override
    protected void onPause() {
        super.onPause();
        if (videoView.isPlaying()) {
            videoView.pause();
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        videoView.start();
    }
}

File: app/src/main/java/com/example/smart_parking_app/MainActivity.kt
package com.example.smart_parking_app

import android.net.Uri
import android.os.Bundle
import android.widget.VideoView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val videoView = findViewById<VideoView>(R.id.videoView)

        // Method 1: If you named it m.mp4
        // val uri = Uri.parse("android.resource://" + packageName + "/" + R.raw.m)

        // Method 2: Generate URI
        val videoResource = "android.resource://$packageName/${R.raw.M}"
        val uri = Uri.parse(videoResource)

        videoView.setVideoURI(uri)
        videoView.requestFocus()

        videoView.setOnPreparedListener { mediaPlayer ->
            mediaPlayer.isLooping = true // Loop the video
            mediaPlayer.setVolume(0f, 0f) // Mute audio
            videoView.start()

            // Optional: Adjust video scaling
            // mediaPlayer.setVideoScalingMode(MediaPlayer.VIDEO_SCALING_MODE_SCALE_TO_FIT_WITH_CROPPING)
        }

        // Handle errors
        videoView.setOnErrorListener { mp, what, extra ->
            // Log error or show fallback
            // You could show an image instead
            true
        }
    }
}